lib/ai/chat.go

```diff
@@ -0,0 +1,74 @@
+/*
```

Comment:
This file is in a new `ai` module to give space for related code for knowledge management and other things needed to make context-aware reasoning work with cluster data.